frontend/taipy-gui/src/components/Taipy/Progress.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+
+import Box from "@mui/material/Box";
+import CircularProgress from "@mui/material/CircularProgress";
+import LinearProgress from "@mui/material/LinearProgress";
+import Typography from "@mui/material/Typography";
+
+import { useDynamicProperty } from "../../utils/hooks";
+
+interface ProgressBarProps {
+    linear?: boolean; //by default - false
+    showValue?: boolean; //by default - false
+    value?: number; //progress value
+    defaultValue?: number; //default progress value
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear, showValue } = props;
+
+    const value = useDynamicProperty(props.value, props.defaultValue, undefined);
+
```

Comment:
```
if (!render) {
    return null;
}
```

Reply:
Here, should I change `render` to `props.render`? Because, if I'm doing this:
```JavaScript
if (!render) {
    return null;
}
```
It's showing an error that `render` is not defined?